pages/live/main.js

```diff
@@ -69,48 +87,53 @@ var lobbyListInit = async function () {
         default: modeString = "Unknown"; break;
       }
 
-      // Add the lobby entry to the output
-      output += `
-  <div class="lobby-entry marginx">
-    <p class="lobby-name">${safeName}</p>
-    <p class="lobby-description">${modeString} - ${lobby.players.length} player${lobby.players.length === 1 ? "" : "s"}</p>
-    <div class="lobby-players">${playersString}</div>
-    <button class="lobby-button" onclick="joinLobby(\`${encodeURIComponent(name)}\`)">Join Lobby</button>
-  </div>
+      // Add the entry data to its respective element
+      entry.innerHTML = `
+        <p class="lobby-name">${safeName}</p>
+        <p class="lobby-description">${modeString} - ${lobby.players.length} player${lobby.players.length === 1 ? "" : "s"}</p>
+        <div class="lobby-players">${playersString}</div>
+        <button class="lobby-button" onclick="joinLobby(\`${lobbyid}\`)">Join Lobby</button>
       `;
 
     }
 
-    const lobbyList = document.querySelector("#lobby-list");
-    lobbyList.innerHTML = output;
-
   };
-  fetchLobbies();
 
-  // Refresh the lobby list every 5 seconds
-  setInterval(fetchLobbies, 5000);
+  // Redraw lobby list on window resize
+  window.addEventListener("resize", drawLobbies);
+  // Redraw lobby list on search bar input
+  lobbySearch.addEventListener("input", drawLobbies);
 
-  // Handle the search bar
-  const lobbySearch = document.querySelector("#lobby-search");
-  lobbySearch.oninput = function () {
+  /**
+   * Fetches the list of lobbies and calls drawLobbies
+   */
+  const fetchLobbies = async function () {
+
+    // Fetch the list of lobbies
+    lobbies = await (await fetch("/api/lobbies/list")).json();
 
-    const query = lobbySearch.value.trim().toLowerCase();
-    const entries = lobbyList.getElementsByClassName("lobby-entry");
+    // Fetch user avatar links into avatarCache
+    for (const lobbyid in lobbies) {
 
-    for (let i = 0; i < entries.length; i ++) {
+      for (const steamid of lobbies[lobbyid].players) {
 
-      const name = entries[i].getElementsByClassName("lobby-name")[0].innerHTML.toLowerCase();
+        try {
+          const profile = await (await fetch(`/api/users/profile/"${steamid}"`)).json();
+          avatarCache[steamid] = profile.avatar;
+        } catch (e) {
+          avatarCache[steamid] = "../icons/unknown.jpg";
+        }
 
-      // Hide the entry if the name does not contain the query
-      if (name.includes(query)) {
-        entries[i].style.display = "";
-      } else {
-        entries[i].style.display = "none";
       }
-
     }
 
+    drawLobbies();
+
   };
+  fetchLobbies();
+
+  // Refresh the lobby list every 5 seconds
+  setInterval(fetchLobbies, 5000);
```

Comment:
Realistically this'll be fine, but wouldn't it be better to push lobbies through the websocket? instead of the client always yapping about new lobbies? im imagining the current approach could introduce significant overhead on large scale deployments

Reply:
Absolutely, yes. But this approach isn't new to this pull request. The scope of this pr also didn't include optimizations like this, which would require writing new code to handle this with websockets. Basically, you're very right, but cba rn.